packages/tools/src/stateManagement/segmentation/config/segmentationVisibility.ts

```diff
@@ -115,9 +121,11 @@ function getSegmentationVisibility(
     return;
   }
 
-  const { segmentsHidden } = representation;
+  const { segmentsHidden, segmentationId } = representation;
+  const indices = getSegmentationIndices(segmentationId);
 
-  return segmentsHidden.size === 0;
+  // TODO: check indices
+  return segmentsHidden.size !== indices.length;
```

Comment:
I believe that this issue may lead to some bugs in the near future. Could you please fix it? It seems like a simple loop should do the trick.